src/main/java/frc/robot/subsystems/intake/IntakeIOTalonFX.java

```diff
@@ -0,0 +1,148 @@
+    inputs.isDrumIRBlocked = drumIRSensor.get();
+
+    inputs.rightRollerAppliedPercentage = rightRollerMotor.getDutyCycle().getValueAsDouble();
+    inputs.leftRollerAppliedPercentage = leftRollerMotor.getDutyCycle().getValueAsDouble();
+    inputs.drumAppliedPercentage = drumMotor.getDutyCycle().getValueAsDouble();
+
+    inputs.rightRollerStatorCurrentAmps =
+        new double[] {rightRollerMotor.getStatorCurrent().getValueAsDouble()};
+    inputs.leftRollerStatorCurrentAmps =
+        new double[] {leftRollerMotor.getStatorCurrent().getValueAsDouble()};
+    inputs.drumStatorCurrentAmps = new double[] {drumMotor.getStatorCurrent().getValueAsDouble()};
+  }
+
+  @Override
+  public void setRightRollerPower(double percentage) {
+    rightRollerMotor.setControl(new DutyCycleOut(percentage));
+  }
+
+  @Override
+  public void setLeftRollerPower(double percentage) {
+    leftRollerMotor.setControl(new DutyCycleOut(percentage));
+  }
+
+  @Override
+  public void setDrumPower(double percentage) {
+    drumMotor.setControl(new DutyCycleOut(percentage));
+  }
+
+  @Override
+  public void setRightRollerCurrent(double current) {
+    rightRollerMotor.setControl(new TorqueCurrentFOC(current));
+  }
+
+  @Override
+  public void setLeftRollerCurrent(double current) {
+    leftRollerMotor.setControl(new TorqueCurrentFOC(current));
+  }
+
+  @Override
+  public void setDrumCurrent(double current) {
+    drumMotor.setControl(new TorqueCurrentFOC(current));
+  }
+
+  private void configureIntakeMotors() {
+    rightRollerMotor = new TalonFX(IntakeConstants.INTAKE_RIGHT_ROLLER_MOTOR_ID);
+    leftRollerMotor = new TalonFX(IntakeConstants.INTAKE_LEFT_ROLLER_MOTOR_ID);
+    drumMotor = new TalonFX(IntakeConstants.INTAKE_DRUM_MOTOR_ID);
+
+    TalonFXConfiguration rightRollerConfig = new TalonFXConfiguration();
+    TalonFXConfiguration leftRollerConfig = new TalonFXConfiguration();
+    CurrentLimitsConfigs rollerCurrentLimits = new CurrentLimitsConfigs();
+
+    rollerCurrentLimits.SupplyCurrentLimit = IntakeConstants.ROLLERS_CONTINUOUS_CURRENT_LIMIT;
+    rollerCurrentLimits.SupplyCurrentThreshold = IntakeConstants.ROLLERS_PEAK_CURRENT_LIMIT;
+    rollerCurrentLimits.SupplyTimeThreshold = IntakeConstants.ROLLERS_PEAK_CURRENT_DURATION;
+    rollerCurrentLimits.SupplyCurrentLimitEnable = true;
+
+    rightRollerConfig.CurrentLimits = rollerCurrentLimits;
+    leftRollerConfig.CurrentLimits = rollerCurrentLimits;
+
+    rightRollerConfig.MotorOutput.Inverted =
+        IntakeConstants.ROLLERS_MOTOR_INVERTED
+            ? InvertedValue.Clockwise_Positive
+            : InvertedValue.CounterClockwise_Positive;
+
+    leftRollerConfig.MotorOutput.Inverted =
+        IntakeConstants.ROLLERS_MOTOR_INVERTED
+            ? InvertedValue.Clockwise_Positive
+            : InvertedValue.CounterClockwise_Positive;
+
+    rightRollerMotor.getConfigurator().apply(rightRollerConfig);
+    leftRollerMotor.getConfigurator().apply(leftRollerConfig);
+
```

Comment:
You need to also configure the neutral value mode and PID constants for the TalonFX controller, before applying the configuration:

```
config.MotorOutput.NeutralMode = NeutralModeValue.Brake;
config.Slot0.kP = kP.get();
config.Slot0.kI = kI.get();
config.Slot0.kD = kD.get();
config.Slot0.kS = kS.get();
```